### PR TITLE
Hybrid: draw sample

### PR DESCRIPTION
**Pull request by jonahkagan**
Task: #1054

Draw two samples - one for CVR ballots and one for non-CVR ballots.

### CODE REVIEW

**Review by umbernhard — APPROVED** [review]

File: server/api/rounds.py [hunk 1]
```diff
@@ -565,23 +572,35 @@ def draw_sample_for_contest(
...
         # this one. Note that this corresponds to the number of SampledBallotDraws,
         # not SampledBallots.
-        num_previously_sampled = SampledBallotDraw.query.filter_by(
-            contest_id=contest.id
-        ).count()
+        num_previously_sampled = (
+            SampledBallotDraw.query.filter_by(contest_id=contest.id)
+            .join(SampledBallot)
+            .join(Batch)
+            .filter_by(has_cvrs=filter_has_cvrs)
```
Comment (benadida) [inline server/api/rounds.py:579]: just checking: you explicitly want this to select for `has_cvrs = null` in the case of ballot comparison and batch comparison?

File: server/api/rounds.py [hunk 2]
```diff
@@ -565,23 +572,35 @@ def draw_sample_for_contest(
...
+            .count()
+        )
 
         # Create the pool of ballots to sample (aka manifest) by combining the
         # manifests from every jurisdiction in the contest's universe.
         manifest = {
             batch_id_to_key[batch.id]: list(range(1, batch.num_ballots + 1))
             for jurisdiction in contest.jurisdictions
             for batch in jurisdiction.batches
+            if batch.has_cvrs == filter_has_cvrs
```
Comment (benadida) [inline server/api/rounds.py:589]: seems consistent with above point, so so far so good :)

**Review by benadida — APPROVED** [review]
one question about filtering, which I'm pretty sure you got right, but worth confirming.

^^ may want to add a comment in the data model about `has_cvrs`, when is it null, specifically that it seems to be null in a ballot comparison audit even though, of course, there are CVRs.

File: server/api/rounds.py (hunk 1 above)
Comment (jonahkagan) [reply server/api/rounds.py:579]: yes, will add a comment

**Pull request merged**